module-system/sov-state/README.md

```diff
@@ -0,0 +1,82 @@
+# sov-state
+This crate provides abstractions specifically designed for storing and retrieving data from a permanent storage, tailored to be used within the Sovereign `module-system`.
+
+## High level explanation:
+At a high level, the crate offers two main abstractions that module developers can utilize to access data:
+
+1. `StateValue`: Is used to store a single value in the state. It provides methods to set a value and retrieve it later.
+1. `StateMap`:  Is used to store mappings in the state. It allows module developers to associate keys with values and retrieve them accordingly
+
+In the future, the crate aims to introduce additional abstractions, such as `StateVec`, to further enhance capabilities of data storage within the module system.
+
+Here is a snippet showcasing part of the `StateValue` API:
+```Rust
+impl StateValue<V> {
+
+    /// Sets a value in the StateValue.
+    pub fn set<S: Storage>(&self, value: V, working_set: &mut WorkingSet<S>) {
+        // Implementation details
+    }
+
+    /// Gets a value from the StateValue or None if the value is absent.
+    pub fn get<S: Storage>(&self, working_set: &mut WorkingSet<S>) -> Option<V> {
+       // Implementation details
+    }
+
+    // Additional methods
+    // ...
+}
+```
+
+
+Both `get` and `set `methods require a `WorkingSet` parameter, which acts as a wrapper around a `key-value` store with additional caching layers.
+
+Module developers can interact with the `WorkingSet`, `StateValue`, and `StateMap`  without worrying about the inner workings of these components. Instead, they can treat them as black boxes that handle the storage and retrieval of data. 
+
+The above API is used in the following way:
+
+```Rust
+state.value.set(some_value, working_set);
+let maybe_value = state.value.get(working_set);
+
+```
+
+## Low level explanation:
+It's important to note that this section is not necessarily required for efficient usage of the `sov-state`.
+
+### `Native` & `Zkp` execution: 
+During `Native` execution, the data is stored in a `key-value` store, which is accessed through the `WorkingSet`. It's worth mentioning that the actual storage mechanism, such as `RocksDB`, is only accessible during this phase when the full node executes the transaction and updates the state.
```

Comment:
nit: Can we add a quick explanation that working sets *read* data from the backing store but write to an in-memory map which can be reverted or committed?

Reply:
Added  relevant info to the `WorkingSet` section